x-pack/plugins/apm/public/components/shared/PropertiesTable/tabConfig.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { get, indexBy, uniq } from 'lodash';
+import { first, has } from 'lodash';
+import { StringMap } from 'x-pack/plugins/apm/typings/common';
+import { APMError } from 'x-pack/plugins/apm/typings/es_schemas/Error';
+import { Transaction } from 'x-pack/plugins/apm/typings/es_schemas/Transaction';
+
+export type TabKey =
+  | keyof Transaction
+  | keyof APMError
+  | 'transaction.custom'
+  | 'error.custom';
+
+export interface Tab {
+  key: TabKey;
+  label: string;
+}
+
+interface TabConfig extends Tab {
+  required: boolean;
+  presortedKeys: string[];
+}
+
+export function getTabsFromObject(obj: Transaction | APMError): Tab[] {
+  return TAB_CONFIG.filter(
+    ({ key, required }) => required || has(obj, key)
+  ).map(({ key, label }) => ({ key, label }));
+}
+
+interface GenericTab {
+  key: string;
+  label: string;
+}
+
+export type KeySorter = (data: StringMap, parentKey?: string) => string[];
+
+export const sortKeysByConfig: KeySorter = (object, currentKey) => {
+  const indexedPropertyConfig = indexBy(TAB_CONFIG, 'key');
```

Comment:
Not a big deal but do we need to re-index this on every sort? This could maybe be at the bottom of the file as INDEXED_TAB_CONFIG or something, then just referenced here. But I doubt it's expensive to index this.

Reply:
IIRC this is to do with testing, and if we want to mock the `TAB_CONFIG` it's easier if it's in here - but I totally agree with you. I'll look into moving it out while keeping the tests mostly as-is.